site/docs/v1/tech/troubleshooting.adoc

```diff
@@ -153,3 +153,26 @@ You can do the former with the `fusionauth-app.memory` setting. See the link:/do
 
 You may do the latter by running a larger server or running fewer competing applications. In particular, applications used by FusionAuth such as Elasticsearch or the database may be moved off to external servers.
 
+=== MapperParsingException
+
+If you are using the Elasticsearch search engine, attributes in the `user.data` and `registration.data` are indexed for you. The data fields can contain arbitrary JSON objects.
+
+If your user search is failing mysteriously and you see an error like this one in your system logs:
+
+```
+org.elasticsearch.index.mapper.MapperParsingException: Could not dynamically add mapping for field [field.email]. Existing mapping for [data.field] must be of type object but found [text].
+```
+
+This means that the type of a field in one of the data fields was changed. In this case, `data.field` was originally text, but the request causing this error message was trying to put an object into that attribute.
+
+Elasticsearch will dynamically build a schema based upon custom data, so it is critical that the schema not be modified.
+
+The fix is:
+
+* understand how the field types changed
+* to modify the problematic objects via the API to remove the mismatched fields 
+* run a re-index operation. 
```

Comment:
Did you you test these steps, not sure if this will work.

The problem is that each time we try to update the user the index request will fail which will blow out the transaction. 

In some customer scenarios I've had to manually update the db. Perhaps one option would be to switch to db search type, fix the users via the API, switch back to ES and then re-index. 

If you have tested this and it works, then I'm good with it, assuming you would hit an exception.

Reply:
I did test it. Running 

```
curl -XPATCH -H "Authorization: $API_KEY" -H "Content-Type: application/json" -d '{"user": {"data": {"foo": "bar"}} }' http://localhost:9011/api/user/b8f1ad7d-def0-48d1-a983-aeabf0122b90
```

where `data.foo` was a string for all the other users, worked and the user was reindexed properly without even a manual reindex.